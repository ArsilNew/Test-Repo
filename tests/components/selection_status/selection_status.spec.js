import React from "react";
import { shallow } from "../Utils";

import SelectionStatus from "../../../src/components/selection_status/selection_status";

describe("SelectionStatus", () => {
  test("default snapshot", () => {
    const { tree } = shallow(<SelectionStatus />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with custom noneSelectedMessage", () => {
    const { tree } = shallow(
      <SelectionStatus noneSelectedMessage="custom no selection message" />
    );
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with selected item", () => {
    const { tree } = shallow(<SelectionStatus selected={[1, 2, 3]} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with selected items and custom clearAllMessage", () => {
    const { tree } = shallow(
      <SelectionStatus selected={[1, 2, 3]} clearAllMessage="custom message" />
    );
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with selected items and custom selectedMessage", () => {
    const { tree } = shallow(
      <SelectionStatus
        selected={[1, 2, 3]}
        selectedMessage="custom selected message"
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
