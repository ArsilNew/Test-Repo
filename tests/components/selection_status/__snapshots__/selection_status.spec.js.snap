// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionStatus default snapshot 1`] = `
<div
  className="selection_status"
>
  <div
    className="status"
  >
    None Selected
  </div>
  <div
    className="clear_all"
    onClick={undefined}
  >
    
  </div>
</div>
`;

exports[`SelectionStatus snapshot with custom noneSelectedMessage 1`] = `
<div
  className="selection_status"
>
  <div
    className="status"
  >
    custom no selection message
  </div>
  <div
    className="clear_all"
    onClick={undefined}
  >
    
  </div>
</div>
`;

exports[`SelectionStatus snapshot with selected item 1`] = `
<div
  className="selection_status"
>
  <div
    className="status"
  >
    3 selected
  </div>
  <div
    className="clear_all"
    onClick={undefined}
  >
    Clear All
  </div>
</div>
`;

exports[`SelectionStatus snapshot with selected items and custom clearAllMessage 1`] = `
<div
  className="selection_status"
>
  <div
    className="status"
  >
    3 selected
  </div>
  <div
    className="clear_all"
    onClick={undefined}
  >
    custom message
  </div>
</div>
`;

exports[`SelectionStatus snapshot with selected items and custom selectedMessage 1`] = `
<div
  className="selection_status"
>
  <div
    className="status"
  >
    3 custom selected message
  </div>
  <div
    className="clear_all"
    onClick={undefined}
  >
    Clear All
  </div>
</div>
`;
